### DISCUSSION TITLE
Dealing with optional and nullable query params

### DISCUSSION THREAD

**Discussion by AFO-UYI** [fastapi/fastapi#8700] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# First approach
@entity_router.get('', responses={
    200: {
        'model': List[EntityInfo]
    }
}, response_model=List[EntityInfo])
async def search_entity(entity_search_params: SearchEntity = Depends()):
    return await search_entity(entity_search_params)


# second approach
@entity_router.get('', responses={
    200: {
        'model': List[EntityInfo]
    }
}, response_model=List[EntityInfo])
async def search_entity(request: Request):
    entity_search_params = SearchEntity(**dict(request.query_params))
    return await search_entity(entity_search_params)
```

### Description

I have an endpoint that searchs entities on database according to the given search params. The database has nullable columns setting the case for searching where an unset param is different to a setted as None param.

For http methods where data is passed through the body, Pydantic class is instanced nicely, because I can use the `pydantic_object.dict(exclude_unset=True) to get the proper and instanced info. So usefull specially for PUT or PATCH endpoints if i need to know if an attribute must be modified to null or if it wasnt to be modified.

But, when I use a pydantic class as query parameter schema using the first approach of the example code, apparently the dependencies solver instance the pydantic class with all the params, included the default ones, making `.dict(exclude_unset=True)` useless.

With the second approach of the example code it works nicely, but I lost the schema definition on OpenAPI if I quit `entity_search_params: SearchEntity = Depends()` because i dont use it. But if i keep it everythign is ok (just a little ugly tho).

I dont know if can be handy that in fastapi/dependencies/utils.py where the value variable is being filled could know better if the dependency is a Pydantic class, in order to preform something near to `PydanticClass(**dict(request.query_params))`. 

Thanks!

### Operating System

Windows, Other

### Operating System Details

The other is the latest tag for python docker image

### FastAPI Version

0.70.0

### Python Version

3.8.1

### Additional Context

starlette==0.16.0
uvicorn==0.15.0

**Comment by gambolputty** [comment]
The reason seams to be that Pydantic does not work with GET query parameters, see https://github.com/tiangolo/fastapi/issues/318#issuecomment-504164334
